### PR TITLE
feat: add support to use existing SM instance for DA

### PR DESCRIPTION
**Pull request by akocbek**
### Description

Issue: https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager/issues/107

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by akocbek** [comment]
```

 2024/04/26 15:02:49 Terraform apply | Error: [ERROR] Error when creating resource instance: Please contact the Service Provider for this error. [422, Unprocessable Entity] This account reached the number of Trial plan instances that are allowed. To proceed, you must select the Standard plan or delete an existing Trial instance. If you delete the Trial plan instance, you must also delete its reclamation by using the IBM Cloud CLI. Learn more: https://ibm.biz/BdPa49 with resp code: {
 ``` 
region: eu-de

**Comment by vburckhardt** [comment]
@ocofaigh - is there a need to clean up trial instances?

File: solutions/standard/variables.tf [hunk 1]
```diff
@@ -41,6 +41,12 @@ variable "secrets_manager_instance_name" {
   default     = "base-security-services-sm"
 }
 
+variable "existing_secrets_manager_crn" {
+  type        = string
+  description = "The CRN of an existing KMS key to use for Secrets Manager. If not supplied, a new key ring and key will be created."
```
Comment (ocofaigh) [inline solutions/standard/variables.tf:46]: The description is wrong here.

**Review by ocofaigh — CHANGES_REQUESTED** [review]
just 1 comment

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**